### PR TITLE
db: refactoring

### LINKED ISSUE #5: Refactor SQL queries
**[Original report](https://bitbucket.org/PascalRD/nemu/issue/19) by Boris Stepanenko (Bitbucket: [GrafIn1992](https://bitbucket.org/GrafIn1992), ).**

----------------------------------------

* They all should be placed in nm\_database.h
* All queries should be consistently named
* Queries order in file should be somehow structured

### PR DESCRIPTION
**Pull request by 0x501D**
Closes #5

### CODE REVIEW

File: src/nm_add_vm.c [hunk 1]
```diff
@@ -390,33 +385,29 @@ void nm_add_vm_to_db(nm_vm_t *vm, uint64_t mac,
         vm->usb_enable ? NM_ENABLE : NM_DISABLE, /* USB enabled */
         NM_DEFAULT_USBVER, /* set USB 3.0 by default */
         NM_DISABLE, /* disable 9pfs by default */
+        NM_DISABLE,
```
Comment (grafin) [inline src/nm_add_vm.c:388]: Please add a comment what this disables (like others).

File: src/nm_database.c [hunk 2]
```diff
@@ -125,6 +122,27 @@ void nm_db_select(const char *query, nm_vect_t *v)
     }
 }
 
+void nm_db_select_value(const char *query, nm_str_t *res)
```
Comment (grafin) [inline src/nm_database.c:125]: This function is very similar to `nm_db_select`, can you reuse some of the code somehow? Maybe create a common `nm_db_select(const char *query, int (*cb)(void, int, char**, char **), void *res)`?

**Review by grafin — CHANGES_REQUESTED** [review]
Thanks for the patches! Great work! Just two minor comments from me)

File: src/nm_add_vm.c (hunk 1 above)
Comment (0x501D) [reply src/nm_add_vm.c]: Added

**Review by grafin — APPROVED** [review]
Thanks for the fixes! LGTM

**Pull request merged**